### DISCUSSION TITLE
Best Practices For Storing Encrypted QRs?

### DISCUSSION THREAD

**Discussion by 2oh1** [selfcustody/krux#301] [Q&A]
I've been thinking a lot about long term security.  Obviously, the first thing to do when setting up a Bitcoin wallet is write down the seed words on paper, make a metal backup, and store those two items securely.  But here's the rub.  Most people don't have access to multiple locations in multiple cities to store those items separately.

We've all seen news footage of tornados leveling a small town or earthquakes doing even worse.  This has me thinking about the safety of storing encrypted QRs.

In Krux, I have PBKDF2 Iterations set at 100,000 and the Encryption Mode set to AES-CBC.

Assuming I'm using a QR encryption key that is at least 25 characters which only I know, is it safe to store the encrypted QR in an iCloud Drive account, which itself is encrypted end to end?

P.S.  The documentation on the main [About Krux ](https://selfcustody.github.io/krux/getting-started/) pages is great.  I'd love to see a Best Practices & Recommendations section get added at some point.  I have to assume you're going to see tons of new users in 2024.

**Comment by odudex** [comment]
Well, encrypted mnemonic QR codes is a new and unique to Krux feature. Being unique to Krux is an advantage for now, in case someone has access to it and tries to scan with widely used signers it won't show anything meaningful, however the small Krux users base means it was not yet tested by many people, so I would suggest to be careful. A Brazilian Bitcoin content creator, Bitdov, created a online [challenge to decrypt a Krux encrypted mnemonic](https://twitter.com/bitdov/status/1676400753261305856), it is published online, was saw probably by some thousands of Bitcoiners and wasn't withdraw, however we can't claim it's safe to be kept in cloud servers.
All we can say it's an additional safety layer, which allow to split a secret in parts and many creative ways to store or transport a mnemonic. Being an image format allow you to disguise it in a picture or movie for example(on beta version you can export to SD as an image file).
Security has many variables, each individual will have his own ideal setup depending on location, how exposed he is, knowledge etc.
To encrypted your mnemonic is to split your secret in two parts, as it is when you use a passphrase, I believe the greatest threat is lose one of the parts, so always have this in mind. Digital storage could be used as a redundancy in some cases, but nothing replaces a physical backup.
For me the encrypted mnemonic QR code could be good redundancy copy of your mnemonic, which you could leave in a more accessible place, as it is less vulnerable than a "naked" plain mnemonic.
This discussion is excellent and I hope more people join it so we can better determine use cases, good practices, risks and this way improve documentation or even the feature itself. I hope more people share their thoughts and ideas about it.

**Comment by jdlcdl** [comment]
I like how simply you've described this as "splitting secrets", and it helps me realize that it's so common to want to "split" secrets into a weak one and a strong one (a weak key we can easily remember, and a cryptographically strong ciphertext that might as well be as weak as the key).   It's important that users are aware when they're doing this.  I'm all for trying to inform the user when the key chosen is too weak, but I have no clue how best to measure it or where to draw the line... maybe "This encrypted mnemonic is as weak as its key".
